test/pkg/environment/common/environment.go

```diff
@@ -115,10 +115,10 @@ func NewClient(ctx context.Context, config *rest.Config) client.Client {
 	}))
 	lo.Must0(cache.IndexField(ctx, &corev1.Node{}, "spec.taints[*].karpenter.sh/disrupted", func(o client.Object) []string {
 		node := o.(*corev1.Node)
-		t, _ := lo.Find(node.Spec.Taints, func(t corev1.Taint) bool {
+		_, found := lo.Find(node.Spec.Taints, func(t corev1.Taint) bool {
 			return t.Key == karpv1.DisruptedTaintKey
 		})
-		return []string{t.Value}
+		return []string{lo.Ternary(found, "true", "false")}
```

Comment:
```suggestion
		return []string{""}
```

nit: I'm fine with this but I think this could also be a string array that contains the empty string if you wanted to just set this to what the actual value was

Reply:
I'm planning on keeping as is, since we don't care about the values for our e2es. Should be easy to change after the fact.